benches/main.rs

```diff
@@ -15,10 +15,11 @@ mod volatile;
 use volatile::benchmark_for_volatile;
 
 #[cfg(feature = "backend-mmap")]
-pub fn create_guest_memory_mmap(size: u64, count: u64) -> GuestMemoryMmap {
+fn create_guest_memory_mmap(size: usize, count: usize) -> GuestMemoryMmap {
     let mut regions: Vec<(GuestAddress, usize)> = Vec::new();
     for i in 0..count {
-        regions.push((GuestAddress(i * size), size as usize));
+        // Safe conversion to `u64`, since it does not truncate.
```

Comment:
I don't think that an `(usize * usize) as u64` can ever be safe, or even `(usize * whatever_type) as u64` (on 64-bit platforms especially) since it can overflow; using `as` will result in panicking if we try to multiply with overflow. We can use `checked_mul()` or something similar here to stay on the safe side. Also, I don't think the `usize -> u64` change of the `count` type in the function signature is necessary useful. Since you need an `u64` here you can just `size as u64 * count` (which is, of course, unsafe).

Reply:
Since this function is used only for benches, I would avoid investing too much effort in handling overflows and maybe the unwrap when `GuestMemoryMmap` is created from ranges. We must not have undefined behavior when using this code, which can happen while converting, but panicking due to improper usage in the benches code, can be directly detected if it happens, so I would not stress about it too much. Do you agree?

Related to the second point, I think you meant `u64 -> usize` change from the signature of the `count` parameter. If this is the case, I agree we can keep `count` as `u64` and only convert `size` from `usize` to `u64` when we form the `GuestAddress`. WDYT?

Reply:
> Since this function is used only for benches, I would avoid investing too much effort in handling overflows and maybe the unwrap when `GuestMemoryMmap` is created from ranges. We must not have undefined behavior when using this code, which can happen while converting, but panicking due to improper usage in the benches code, can be directly detected if it happens, so I would not stress about it too much. Do you agree?

:+1: the comment should be updated though 
> Related to the second point, I think you meant `u64 -> usize` change from the signature of the `count` parameter. If this is the case, I agree we can keep `count` as `u64` and only convert `size` from `usize` to `u64` when we form the `GuestAddress`. WDYT?

Yes, that's what I meant.

Reply:
The comment is still not rephrased/removed.